File: wela_analytics/student_grades/report/student_grade_report/student_grade_report.js
// Copyright (c) 2016, Marc Anthony Reyes and contributors
// For license information, please see license.txt
/* eslint-disable */

/*frappe.query_reports["Student_Grade_Report"] = {
	"filters": [
		{
			"fieldname": "full_name",
			"label": __("Full Name"),
			"fieldtype": "Data",
			"reqd": 0,
		},
		{
			"fieldname": "eng_gr",
			"label": __("English"),
			"fieldtype": "Int",
			"reqd": 0,
		},
		{
			"fieldname": "mat_gr",
			"label": __("Math"),
			"fieldtype": "Int",
			"reqd": 0,
		},
		{
			"fieldname": "fil_gr",
			"label": __("Filipino"),
			"fieldtype": "Int",
			"reqd": 0,
		},
		{
			"fieldname": "sci_gr",
			"label": __("Science"),
			"fieldtype": "Int",
			"reqd": 0,
		},
		{
			"fieldname": "ss_gr",
			"label": __("Social Studies"),
			"fieldtype": "Int",
			"reqd": 0,
		},
		{
			"fieldname": "avg_gr",
			"label": __("Average"),
			"fieldtype": "Int",
			"reqd": 0,
		}
	]
}
*/

frappe.query_reports["Student_Grade_Report"] = {
	"filters": [
		{
			"fieldname": "qtr",
			"label": __("Quarter"),
			"fieldtype": "Link",
			"options": "Grade Quarter",
			"reqd": 0,
			"width": 20,
		}
	]
}

File: wela_analytics/student_grades/report/student_grade_report/student_grade_report.py
# Copyright (c) 2013, Marc Anthony Reyes and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe


def execute(filters=None):
    columns, data = get_student_list(filters.get('qtr'))
    return columns, data


def get_student_list(qtr):
    columns = [
        {
            "label": "Full Name",
            "width": 300,
            "fieldname": "full_name"
        },
        {
            "label": "Quarter",
            "width": 100,
            "fieldname": "qtr"
        },
        {
            "label": "English Grade",
            "width": 100,
            "fieldname": "eng_gr"
        },
        {
            "label": "Math Grade",
            "width": 100,
            "fieldname": "mat_gr"
        },
        {
            "label": "Filpino Grade",
            "width": 100,
            "fieldname": "fil_gr"
        },
        {
            "label": "Science Grade",
            "width": 100,
            "fieldname": "sci_gr"
        },
        {
            "label": "Social Studies Grade",
            "width": 100,
            "fieldname": "ss_gr"
        },
        {
            "label": "Average Grade",
            "width": 100,
            "fieldname": "avg_gr"
        }
    ]

    get_student_list_sql = """SELECT full_name, qtr, eng_gr, mat_gr, fil_gr, sci_gr, ss_gr, avg_gr
                                FROM `tabStudent Grade`"""

    if qtr:
        get_student_list_sql = """SELECT full_name, qtr, eng_gr, mat_gr, fil_gr, sci_gr, ss_gr, avg_gr
                                FROM `tabStudent Grade` {where}"""

        get_student_list_sql = get_student_list_sql.replace(
            "{where}", "WHERE qtr='{}'".format(qtr))

    result = frappe.db.sql(get_student_list_sql, as_dict=True)
    return columns, result
